clients/venice-producer/src/main/java/com/linkedin/venice/producer/VeniceProducerMetrics.java

```diff
@@ -0,0 +1,85 @@
+package com.linkedin.venice.producer;
+
+import com.linkedin.venice.stats.AbstractVeniceStats;
+import io.tehuti.metrics.MetricsRepository;
+import io.tehuti.metrics.Sensor;
+import io.tehuti.metrics.stats.Avg;
+import io.tehuti.metrics.stats.Max;
+import io.tehuti.metrics.stats.OccurrenceRate;
+import io.tehuti.metrics.stats.Total;
+
+
+public class VeniceProducerMetrics extends AbstractVeniceStats {
+  private final boolean enableMetrics;
+  private Sensor requestSensor = null;
+  private Sensor putRequestSensor = null;
+  private Sensor deleteRequestSensor = null;
+  private Sensor updateRequestSensor = null;
+  private Sensor successRequestSensor = null;
+  private Sensor failedRequestSensor = null;
+  private Sensor produceLatencySensor = null;
+  private Sensor pendingRequestSensor = null;
+
+  public VeniceProducerMetrics(MetricsRepository metricsRepository, String storeName) {
+    super(metricsRepository, storeName);
+
+    if (metricsRepository != null) {
+      enableMetrics = true;
+      requestSensor = registerSensor("write_request", new OccurrenceRate());
+      putRequestSensor = registerSensor("put_request", new OccurrenceRate());
+      deleteRequestSensor = registerSensor("delete_request", new OccurrenceRate());
+      updateRequestSensor = registerSensor("update_request", new OccurrenceRate());
+
+      successRequestSensor = registerSensor("success_write_request", new OccurrenceRate());
+      failedRequestSensor = registerSensor("failed_write_request", new OccurrenceRate());
```

Comment:
Hmm... I would prefer avoiding the term "request" as it seems to imply a meaning of transactionality. How about replacing "request" with "operation" or with "event"?